connect/runtime/src/test/java/org/apache/kafka/connect/runtime/isolation/PluginsTest.java

```diff
@@ -185,6 +181,102 @@ public void shouldInstantiateAndConfigureDefaultHeaderConverter() {
         assertTrue(headerConverter instanceof SimpleHeaderConverter);
     }
 
+    @Test(expected = ConnectException.class)
+    public void shouldThrowIfPluginThrows() {
+        TestPlugins.assertInitialized();
+
+        plugins.newPlugin(
+            TestPlugins.ALWAYS_THROW_EXCEPTION,
+            new AbstractConfig(new ConfigDef(), Collections.emptyMap()),
+            Converter.class
+        );
+    }
+
+    @Test
+    public void shouldShareStaticValuesBetweenSamePlugin() {
+        // Plugins are not isolated from other instances of their own class.
+        TestPlugins.assertInitialized();
+        Converter firstPlugin = plugins.newPlugin(
+            TestPlugins.SAMPLING,
+            new AbstractConfig(new ConfigDef(), Collections.emptyMap()),
+            Converter.class
+        );
+
+        assertTrue(firstPlugin instanceof SamplingTestPlugin);
+
+        assertEquals(1, ((SamplingTestPlugin) firstPlugin).dynamicInitializations());
```

Comment:
what 1 and 2 mean will be hard to follow after we review this PR. 

How about naming such expectations as: `EXPECTED_NUMBER_FOOBAR` etc appropriately?